eslint.config.cjs

```diff
@@ -0,0 +1,166 @@
+const typescriptEslint = require('@typescript-eslint/eslint-plugin')
+const jest = require('eslint-plugin-jest')
+const simpleImportSort = require('eslint-plugin-simple-import-sort')
+const _import = require('eslint-plugin-import')
+const localRules = require('eslint-plugin-local-rules')
+const path = require('path')
+
+const { fixupPluginRules } = require('@eslint/compat')
+
+const globals = require('globals')
+const tsParser = require('@typescript-eslint/parser')
+const js = require('@eslint/js')
+
+const { FlatCompat } = require('@eslint/eslintrc')
+
+const compat = new FlatCompat({
+  baseDirectory: __dirname,
+  recommendedConfig: js.configs.recommended,
+  allConfig: js.configs.all,
+})
+
+const project = path.resolve(__dirname, 'tsconfig.json')
+
+module.exports = [
+  {
+    files: ['**/*.ts'],
+  },
+  {
+    ignores: [
+        ".prisma",
+        ".github",
+          ".github/renovate.json",
+          "**/dist/**",
+          "**/esm/**",
+          "**/build/**",
+          "**/fixtures/**",
+          "**/byline.ts",
+          "**/prism.ts",
+          "**/charm.ts",
+          "**/pnpm-lock.yaml",
+          "**/generated-dmmf.ts",
+          "**/packages/client/generator-build/**",
+          "**/packages/client/declaration/**",
+          "**/packages/client/runtime/**",
+          "**/packages/client/src/__tests__/types/**",
+          "packages/client/scripts/default-index.js",
+      '**/packages/cli/prisma-client/**',
+          "**/packages/cli/install/**",
+          "**/packages/cli/preinstall/**",
+          "packages/cli/**/tmp-*",
+          "**/sandbox/**",
+    ],
+  },
+  ...compat.extends(
+    'eslint:recommended',
+    'plugin:@typescript-eslint/eslint-recommended',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:@typescript-eslint/recommended-requiring-type-checking',
+    'plugin:prettier/recommended',
+    'plugin:jest/recommended',
+  ),
+  {
+    plugins: {
+      '@typescript-eslint': typescriptEslint,
+      jest,
+      'simple-import-sort': simpleImportSort,
+      import: fixupPluginRules(_import),
```

Comment:
Oh! That's interesting, because I identified this as a blocker recently and this unblocks it.

(In https://github.com/prisma/prisma/pull/24148#pullrequestreview-2052063099)